File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_py/code_output/FitnessTracker.py
class FitnessTracker:
    def __init__(self, height, weight, age, sex):
        self.height = height
        self.weight = weight
        self.age = age
        self.sex = sex
        self.BMI_std = [[20, 25], [19, 24]]

    def get_bmi(self):
        return self.weight / (self.height ** 2)

    def condition_judge(self):
        bmi = self.get_bmi()
        index = 0 if self.sex == "male" else 1
        if bmi > self.BMI_std[index][1]:
            return 1
        elif bmi < self.BMI_std[index][0]:
            return -1
        else:
            return 0

    def calculate_calorie_intake(self):
        if self.sex == "male":
            bmr = 10 * self.weight + 6.25 * self.height - 5 * self.age + 5
        else:
            bmr = 10 * self.weight + 6.25 * self.height - 5 * self.age - 161

        condition = self.condition_judge()
        if condition == 1:
            return bmr * 1.2
        elif condition == -1:
            return bmr * 1.6
        else:
            return bmr * 1.4

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_cpp/code_output/FitnessTracker.cpp
#include <string>

class FitnessTracker {
private:
    double height;
    double weight;
    int age;
    std::string sex;
    double BMI_std[2][2] = {{20, 25}, {19, 24}};

public:
    FitnessTracker(double height, double weight, int age, std::string sex)
        : height(height), weight(weight), age(age), sex(sex) {}

    double getBMI() {
        return weight / (height * height);
    }

    int conditionJudge() {
        double BMI = getBMI();
        int index = (sex == "male") ? 0 : 1;
        if (BMI > BMI_std[index][1]) {
            return 1;
        }
        else if (BMI < BMI_std[index][0]) {
            return -1;
        }
        else {
            return 0;
        }
    }

    double calculateCalorieIntake() {
        double BMR;
        if (sex == "male") {
            BMR = 10 * weight + 6.25 * height - 5 * age + 5;
        }
        else {
            BMR = 10 * weight + 6.25 * height - 5 * age - 161;
        }

        int condition = conditionJudge();
        if (condition == 1) {
            return BMR * 1.2;
        }
        else if (condition == -1) {
            return BMR * 1.6;
        }
        else {
            return BMR * 1.4;
        }
    }
};